consensus/polybft/bridge_event_manager.go

```diff
@@ -381,57 +393,66 @@ func (b *bridgeEventManager) PostEpoch(req *PostEpochRequest) error {
 
 	b.nextBridgeEventIDIndex = nextBridgeEventIDIndex
 
-	b.lock.Unlock()
-
-	return b.buildBridgeBatch(req.DBTx)
-}
-
-// PostBlock notifies bridge event manager that a block was finalized,.
-// Additionally, it will remove any processed bridge message events.
-func (b *bridgeEventManager) PostBlock(req *PostBlockRequest) error {
-	bridgeBatch, err := getBridgeBatchSignedTx(req.FullBlock.Block.Transactions)
+	b.nextBridgeEventIDInternal, err = req.SystemState.GetNextCommittedIndexInternal(b.internalChainID)
 	if err != nil {
+		b.lock.Unlock()
+
 		return err
 	}
 
-	// no bridge batch message -> this is not end of sprint block
-	if bridgeBatch == nil {
-		return nil
-	}
+	b.lock.Unlock()
 
-	if err := b.state.BridgeMessageStore.insertBridgeBatchMessage(bridgeBatch, req.DBTx); err != nil {
-		return fmt.Errorf("insert batch message error: %w", err)
+	if err := b.buildBridgeBatch(req.DBTx, true); err != nil {
+		return err
 	}
 
-	b.lock.Lock()
-	defer b.lock.Unlock()
+	return b.buildBridgeBatch(req.DBTx, false)
+}
 
-	length := len(bridgeBatch.MessageBatch.Messages)
-	// update the nextBridgeEventIdIndex since a bridge batch was submitted
-	b.nextBridgeEventIDIndex =
-		bridgeBatch.MessageBatch.Messages[length-1].ID.Uint64() + 1
-	// batch was submitted, so discard what we have in memory, so we can build a new one
-	b.pendingBridgeBatches = nil
+// CreateInternalBatch creates batch from internal events.
+func (b *bridgeEventManager) CreateInternalBatch() error {
+	if err := b.buildBridgeBatch(nil, true); err != nil {
+		// we don't return an error here. If bridge message event is inserted in db,
+		// we will just try to build a batch on next block or next event arrival
+		b.logger.Error("could not build a batch on arrival of new bridge message event",
+			"err", err)
+	}
 
 	return nil
 }
 
 // buildBridgeBatch builds a new bridge batch, signs it and gossips its vote for it
-func (b *bridgeEventManager) buildBridgeBatch(dbTx *bolt.Tx) error {
+func (b *bridgeEventManager) buildBridgeBatch(dbTx *bolt.Tx, internal bool) error {
 	if !b.runtime.IsActiveValidator() {
 		// don't build batch if not a validator
 		return nil
 	}
 
+	var (
```

Comment:
II would recommend next approach.

Have three functions:
- `buildExternalBridgeBatch(dbTx *bolt.Tx) error`
- `buildInternalBridgeBatch(dbTx *bolt.Tx) error`
- `buildBridgeBatch(dbTx *bolt.Tx, sourceChainID, destinationChainID, nextBridgeEventIDIndex uint64) error`.

Then, make the first two functions call the third one with appropriate parameters.